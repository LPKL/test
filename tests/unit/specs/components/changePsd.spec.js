import { shallowMount } from '@vue/test-utils'
import mountOption from '~/cores/vueinstance'
import ChangePsd from '@/views/_system/user/person/changePsd.vue'
describe('ChangePsd', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallowMount(ChangePsd, mountOption({
      passwordForm: {
        oldPass: '123456',
        newPass: '123456',
        confirmPass: '123456'
      }
    }))
  })
  test('测试组件内的变量类型', () => {
    const vm = wrapper.vm
    // console.log(vm._data.passwordForm.oldPass)
    vm.$nextTick(() => {
      expect(typeof vm.passwordForm).toMatch('object')
      //   数据为空，单元测试报错
      //   expect(vm.passwordForm.oldPass).toMatch('123456')
      const ele = wrapper.find('.save_button')
      ele.trigger('click')
    })
  })
  test('生成快照', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
