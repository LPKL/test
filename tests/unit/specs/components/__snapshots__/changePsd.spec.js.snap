// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePsd 生成快照 1`] = `
<div
  class="change_container"
  passwordform="[object Object]"
>
  <el-card-stub
    class="card"
  >
    <h3>
      修改密码
    </h3>
     
    <div
      class="form_box"
    >
      <el-form-stub
        labelposition="left"
        labelsuffix=""
        labelwidth="80px"
        model="[object Object]"
        rules="[object Object]"
        showmessage="true"
        validateonrulechange="true"
      >
        <el-form-item-stub
          inlinemessage=""
          label="旧密码"
          prop="oldPass"
          showmessage="true"
        >
          <el-input-stub
            autocomplete="off"
            class="new-input"
            showpassword="true"
            type="password"
            validateevent="true"
            value=""
          />
        </el-form-item-stub>
         
        <el-form-item-stub
          inlinemessage=""
          label="新密码"
          prop="newPass"
          showmessage="true"
        >
          <el-input-stub
            autocomplete="off"
            class="new-input"
            showpassword="true"
            type="password"
            validateevent="true"
            value=""
          />
        </el-form-item-stub>
         
        <el-form-item-stub
          inlinemessage=""
          label="确认密码"
          prop="confirmPass"
          showmessage="true"
        >
          <el-input-stub
            autocomplete="off"
            class="new-input"
            showpassword="true"
            type="password"
            validateevent="true"
            value=""
          />
        </el-form-item-stub>
      </el-form-stub>
       
      <div
        class="save_button"
      >
        <el-button-stub
          class="btn"
          icon=""
          nativetype="button"
          type="primary"
        >
          保存
        </el-button-stub>
      </div>
    </div>
  </el-card-stub>
</div>
`;
